backend/src/nodes/nodes/utility/random_number.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import annotations
+from random import randint, seed
+from . import category as UtilityCategory
+
+from ...node_base import NodeBase, group
+from ...node_factory import NodeFactory
+from ...properties.inputs import NumberInput
+from ...properties.outputs import NumberOutput
+
+
+@NodeFactory.register("chainner:utility:random_number")
+class RandomNumberNode(NodeBase):
+    def __init__(self):
+        super().__init__()
+        self.description = "Generate a random integer."
+        self.inputs = [
+            NumberInput(
+                "Minimum Value",
+                minimum=None,
+                maximum=None,
+            ),
+            NumberInput(
+                "Maximum Value",
+                minimum=None,
+                maximum=None,
+                default=100,
+            ),
+            group("seed")(
+                NumberInput(
+                    "Seed",
+                    minimum=0,
+                    maximum=None,
+                ),
+            ),
+            NumberInput(
+                "Index from Iterator",
+                minimum=0,
+                maximum=None,
+            ),

```

Comment:
Using a number input here isn't ideal. Users aren't supposed to enter numbers into this input, they are supposed to connect it to something. We can enforce this by making this an optional generic input.

Please note that this means that `frameval` can now be `None`. Please make sure to denotes this in the type hint.

```suggestion
            BaseInput(input_type="uint", label="Index from Iterator", kind="generic").make_optional(),
```

Reply:
> Using a number input here isn't ideal. Users aren't supposed to enter numbers into this input, they are supposed to connect it to something. We can enforce this by making this an optional generic input.
> 
> Please note that this means that `frameval` can now be `None`. Please make sure to denotes this in the type hint.

Thank you for the suggestions. I have updated it per your recommendation, and in the case of the 'None' for the frameval I change it to 0.

The only other thing I wonder now is would this node be better categorized under the "Value" subheading rather than "Math"? I just now noticed the "Number" node and thought perhaps it would be better to group them together.